### PR TITLE
fix: read header lower-case

### PR DESCRIPTION
**Pull request by marcoieni**
Header names are lower-cased. See [docs](https://nodejs.org/api/http.html#messageheaders):
image

I also run a manual test:
216300787-d27fa2b6-6950-4521-91a2-2bf337949168_png__2342×846_

### CODE REVIEW

File: packages/engine-core/src/__tests__/errors.test.ts [hunk 1]
```diff
@@ -168,7 +168,7 @@ describe('getErrorMessageWithLink', () => {
 
       If you want the Prisma team to look into it, please open the link above 🙏
       To increase the chance of success, please post your schema and a snippet of
-      how you used Prisma Client in the issue. 
+      how you used Prisma Client in the issue.
```
Comment (MarcoIeni) [inline packages/engine-core/src/__tests__/errors.test.ts:171]:
```suggestion
      how you used Prisma Client in the issue. 
```

**Review by SevInf — APPROVED** [review]

**Comment by MarcoIeni** [comment]
Some jobs failed. Is the CI a bit flaky? Or do you think it's a problem with my code?

**Comment by SevInf** [comment]
@MarcoIeni that test is unfortinately known to be flaky, your changes look good

**Comment by MarcoIeni** [comment]
Now it's all green!

**Pull request merged**